lib/mlir/Pipes/MLIRContainer.cpp

```diff
@@ -0,0 +1,270 @@
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <optional>
+
+#include "mlir/Bytecode/BytecodeReader.h"
+#include "mlir/Bytecode/BytecodeWriter.h"
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+#include "mlir/IR/FunctionInterfaces.h"
+#include "mlir/IR/IRMapping.h"
+#include "mlir/IR/SymbolTable.h"
+#include "mlir/Parser/Parser.h"
+
+#include "revng/Pipeline/RegisterContainerFactory.h"
+
+#include "revng-c/Pipes/Kinds.h"
+#include "revng-c/mlir/Pipes/MLIRContainer.h"
+
+#include "MLIRLLVMHelpers.h"
+
+using namespace revng;
+using namespace revng::pipes;
+using namespace revng::mlir_llvm;
+
+static constexpr auto &TheKind = kinds::MLIRLLVMFunctionKind;
+
+using ContextPtr = std::unique_ptr<mlir::MLIRContext>;
+using OwningModuleRef = mlir::OwningOpRef<mlir::ModuleOp>;
+
+static OwningModuleRef cloneModuleInto(mlir::ModuleOp SourceModule,
+                                       mlir::MLIRContext &DestinationContext) {
+  llvm::SmallString<1024> Buffer;
```

Comment:
Consider `using namespace mlir` and/or `llvm`, it should increase readability.

Reply:
I added using declarations for some specific commonly used things but did not add any using directives.